preprocessor/processor.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+import os
+import glob
+import cv2 as cv
+
+allowed_file_extensions = ['.png', '.jpg', '.jpeg']
+
+# recursively load and normalize images using histogram equalization from OpenCV
+def normalize_images(inputdirectory, outputdirectory):
+    inputpath = Path(inputdirectory)
+    outputpath = Path(outputdirectory)
+    inputfiles = []
+
+    for fe in allowed_file_extensions:
+        inputfiles.extend(list(inputpath.glob('**/*' + fe)))
+
+    for f in inputfiles:
+        abs_file_path = Path.absolute(f)
+        file_path = str(abs_file_path)
+        output_file_name = get_output_file_name(f, inputpath, outputpath)
+        print('normalizing image: ' + str(f) + ' to: ' + str(output_file_name))
```

Comment:
Nit: for consistency with `main.py`, it would be good to use string formatting here:

```py
print('normalizing image: {} to: {}'.format(f, output_file_name))
```